# 4. Evaluation

In [21]:
# Model importieren
import joblib
import pandas as pd
import json
import numpy as np

model = joblib.load('popularitaetsmodel.pkl')

# Daten importieren
y_test = pd.read_pickle('y_test.pkl')
X_test = np.load("X_test.npy")

In [24]:
# Evaluierung

from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score

# Vorhersage für Testmenge
y_pred = model.predict(X_test)

# Leistung des Modells bewerten
r2 = r2_score(y_test, y_pred)  # R² Leistung
mae = mean_absolute_error(y_test, y_pred)  # Mean Absolute Error
mse = mean_squared_error(y_test, y_pred)  # Mean Squared Error
rmse = mean_squared_error(y_test, y_pred)  # Root Mean Squared Error (RMSE)

# Bewertungsmetriken drucken
print("Model Performance:")
print(f"R^2 Score: {r2:.4f}")
print(f"Mean Absolute Error (MAE): {mae:.4f}")
print(f"Mean Squared Error (MSE): {mse:.4f}")

Model Performance:
R^2 Score: 0.6611
Mean Absolute Error (MAE): 6.7795
Mean Squared Error (MSE): 81.3018


Wichtig noch anzumerken ist das ich keine Wahrheitsmatrix erstellt habe da ich keine Klassen in meinem Datensatz habe, die ich aufteilen könnte. 

## Modellleistung und Analyse
Das Random Forest-Modell erreichte einen R² Score von 0.66 und einen MAE von ~6.78 , was auf moderate Vorhersagegenauigkeit hinweist. Mögliche Gründe hierfür sind:

Unzureichende Features : Relevante Merkmale könnten fehlen oder nicht optimal vorverarbeitet sein.

Komplexität der Daten : Saisonale Effekte oder externe Einflüsse könnten unberücksichtigt bleiben.

Hyperparameter-Tuning : Suboptimale Parameter könnten die Performance begrenzen.


Zur Verbesserung könnten Feature-Engineering, Cross-Validation und alternative Modelle wie Gradient Boosting eingesetzt werden.